src/System.Text.ValueBuilder/System/Text/ValueStringBuilder.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Text
+{
+    // *** This file is a simple copy from CoreFX/CoreCLR (made public)
+
+    public unsafe ref partial struct ValueStringBuilder
+    {
+        private char[] _arrayToReturnToPool;
+        private Span<char> _chars;
+        private int _pos;
+        private fixed char _default[DefaultBufferSize];
```

Comment:
Why this is not analyzed in constructors? Perhaps it is not needed in `public ValueStringBuilder(Span<char> initialBuffer)` but in `public ValueStringBuilder(int initialCapacity)` we could check that initialCapacity > DefaultBufferSize before allocation.

Reply:
> This file is a simple copy from CoreFX/CoreCLR (made public)

Is it worth all the comments on this file?

Reply:
> You can't allocate on the stack and return to the caller

Yes. My thoughts was that we could allocate only in ToString() or never if we don't call ToStrin().

Reply:
> I'm not sure what you mean? I do need to update the comment for the small changes I've made.

@JeremyKuhne I'm just asking if there's any point in reviewing this file as part of this PR since I'd assume changes should be made in the original file 😄